recipes/rmlui/3.3/conanfile.py

```diff
@@ -0,0 +1,150 @@
+    options = {
+        "enable_rtti_and_exceptions": [True, False],
+        "font_interface": ["freetype", None],
+        "fPIC": [True, False],
+        "shared": [True, False],
+        "with_lua_bindings": [True, False],
+        "with_thirdparty_containers": [True, False]
+    }
+    default_options = {
+        "enable_rtti_and_exceptions": True,
+        "font_interface": "freetype",
+        "fPIC": True,
+        "shared": False,
+        "with_lua_bindings": False,
+        "with_thirdparty_containers": True
+    }
+    build_requires = ["cmake/3.20.0"]
+    exports_sources = ["CMakeLists.txt"]
+    generators = ["cmake", "cmake_find_package"]
+
+    @property
+    def _minimum_compilers_version(self):
+        # Reference: https://en.cppreference.com/w/cpp/compiler_support/14
+        return {
+            "apple-clang": "5.1",
+            "clang": "3.4",
+            "gcc": "5",
+            "intel": "17",
+            "sun-cc": "5.15",
+            "Visual Studio": "15"
+        }
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 14
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compilers_version.get(
+            str(self.settings.compiler))
+        if not min_version:
+            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
+                self.name, self.settings.compiler))
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
+                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
+
+    def requirements(self):
+        if self.options.font_interface == "freetype":
+            self.requires("freetype/2.10.1")
+
+        if self.options.with_lua_bindings:
+            self.requires("lua/5.3.5")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        if not hasattr(self, "_cmake"):
+            self._cmake = CMake(self)
+            self._cmake.definitions["BUILD_LUA_BINDINGS"] = self.options.with_lua_bindings
+            self._cmake.definitions["BUILD_SAMPLES"] = False
+            self._cmake.definitions["DISABLE_RTTI_AND_EXCEPTIONS"] = not self.options.enable_rtti_and_exceptions
+            self._cmake.definitions["ENABLE_PRECOMPILED_HEADERS"] = True
```

Comment:
The only thing I'm left wondering about is whether it is worth to require cmake for building with precompiled headers.
The only benefit precompiled headers have is potential faster compile times. The resulting binary should not differ.

Is it worth it for rmlui?

Reply:
Well.. the answer to that would depend largely on the build environment. Therefore, I checked the build logs to see what happens and noted that each build seems to start off with a clean slate as far as cached packages are concerned.

With this in mind, I made some measurements locally using MSVC, to be able to provide a rough estimate as to the improvement in build time. I used the following commands to build packages and measured until after the test had concluded:
```
conan remove cmake -f
conan create . 3.3@te/st
```

|attempt|with pch|without pch|
|-|-|-|
|1|1:59|2:45|
|2|2:04|2:31|

Based on the above, I'd assume that using precompiled headers would save the build system behind conan-center-index time in case of other compilers as well. Since improvement in build time should be considered a positive outcome in my opinion, I would conclude that it's indeed worth it - unless I'm missing something that would also play a factor in the current context.

Reply:
Thanks for the benchmark!.
A >25% performance increase is indeed impressive.
Though, it should be noted that this cost is only paid once (on the cci servers) or when somebody is building the package themselve.

Reply:
> it should be noted that this cost is only paid once

It's fairly rare that consumers actually build the library... even more rare they build twice... (whole point of Conan is it manages the binaries for you)

For that reason we've avoided these types of things because they were complicated to manage for CCI ... but it's only two lines.

We'll see if the Conan team has any objectives.